hibernate-core/src/main/java/org/hibernate/type/descriptor/sql/JdbcTypeJavaClassMappings.java

```diff
@@ -103,6 +103,7 @@ public Class determineJavaClassForJdbcTypeCode(int typeCode) {
 		workMap.put( BigDecimal.class, Types.NUMERIC );
 		workMap.put( BigInteger.class, Types.NUMERIC );
 		workMap.put( Boolean.class, Types.BIT );
+		workMap.put( Byte.class, Types.TINYINT );
 		workMap.put( Short.class, Types.SMALLINT );
```

Comment:
you might need to update relevant user guide documentation at `documentation/src/main/asciidoc/userguide/chapters/domain/types.adoc` as well.

Reply:
@NathanQingyangXu Nathan, the Basic Types table is already updated. Please take a look at:
documentation\src\main\asciidoc\userguide\chapters\domain\basic_types.adoc